Feature/map markers

marekdovjak opened the pull request.

Commit:
Add temporary sdk initialization until browse map module is ready

Commit:
pokusy s markers manazerom

Commit:
Fix handllig marker highlight state

Commit:
Merge branch 'feature/browse-map' into feature/map_markers

# Conflicts:
#	Example/SygicMapsKit.xcodeproj/project.pbxproj
#	Example/SygicMapsKit/SygicMapsAPIConfig.swift
#	Example/SygicMapsKit/ViewController.swift

Commit:
Improve pin display logic

Commit:
remove key/secret

Commit:
Add documentation, clean unused code, improve highlighting markers

Commit:
Add center map button and functionality

Commit:
Add map selection modes to SYMKBrowseMapViewController. Add some documentation.

Commit:
Remove keys

Commit:
Not needed anymore

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -37,9 +64,24 @@ public class SYMKBrowseMapViewController: UIViewController {
+        guard let view = view as? SYMKBrowserMapView, let mapView = view.mapView else { return }
+        mapSelectionManager.mapObjectsManager = self
+        mapSelectionManager.clusterLayer = SYMapMarkersCluster()
+        mapView.addMapMarkersCluster(self.mapSelectionManager.clusterLayer!)
```
```suggestion
        mapView.addMapMarkersCluster(mapSelectionManager.clusterLayer!)
```

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -37,9 +64,24 @@ public class SYMKBrowseMapViewController: UIViewController {
+    @objc func didTapRecenterButton() {
+        guard let view = view as? SYMKBrowserMapView, let mapView = view.mapView else { return }
+        mapView.cameraMovementMode = .followGpsPositionWithAutozoom
+        mapView.cameraRotationMode = .vehicle
```
toto chceme vzdy nastavovat na vehicle? nechceme tam default lock na sever resp az viacnasobne tapanie na button by prepinalo medzi lock na sever a .vehicle?

marekdovjak replied:
neviem, nemame definovane ako sa to ma spravat, tak som tam zatial dal len taky zaklad.

jcaliksygic committed:
Update SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift

Co-Authored-By: marekdovjak <mdovjak@sygic.com>

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -56,25 +98,74 @@ extension SYMKBrowseMapViewController: SYMapViewDelegate {
+    }
+    
+    public func mapView(_ mapView: SYMapView, didSelect objects: [SYViewObject]) {
+        if mapSelectionMode == .none {
```
na taketo jednoriadkove ify je krajsie pouzit guard mapSelection != .none else { return }

marekdovjak replied:
ok

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -56,25 +98,74 @@ extension SYMKBrowseMapViewController: SYMapViewDelegate {
+        
+        let hadPin = !mapSelectionManager.markers.isEmpty
+        if !mapSelectionManager.markers.isEmpty {
+            self.mapSelectionManager.removeAllMarkers()
```
```suggestion
            mapSelectionManager.removeAllMarkers()
```

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -56,25 +98,74 @@ extension SYMKBrowseMapViewController: SYMapViewDelegate {
+        
+        for obj in objects {
+            if let poi = obj as? SYPoiObject, poi.type == .poi, mapSelectionMode == .all || mapSelectionMode == .all {
+                SYPlaces.shared().loadPoiObjectPlace(poi) { (place: SYPlace) in
```
```suggestion
                SYPlaces.shared().loadPoiObjectPlace(poi) { [unowned self] (place: SYPlace) in
```

marekdovjak replied:
netreba, nenastava tam ziadny retain cyklus

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -56,25 +98,74 @@ extension SYMKBrowseMapViewController: SYMapViewDelegate {
+        
+        if let coord = viewObj?.coordinate {
+            if let pin = SYMKMapPin(coordinate: coord, properties: SYUIPinViewViewModel(icon: SygicIcon.POIPoi, color: .darkGray, selected: true, animated: false)) {
+                self.mapSelectionManager.addMapMarker(pin)
```
```suggestion
                mapSelectionManager.addMapMarker(pin)
```

Kraci commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowserMapView.swift:
```diff
@@ -50,4 +50,11 @@ class SYMKBrowserMapView: UIView {
+        bringSubview(toFront: recenter)
+        recenter.translatesAutoresizingMaskIntoConstraints = false
+        recenter.leadingAnchor.constraint(equalTo: safeLeadingAnchor, constant: 32).isActive = true
+        recenter.bottomAnchor.constraint(equalTo: safeBottomAnchor, constant: -16).isActive = true
```
Preco je toto raz 32 a 16? Tipujem, ze na Xku to vyzera takto lepsie? lebo na nonX iphonoch to vyzera celkom zle. Pre normal iphony by to mohlo byt 16, teda hore vytvorena konstanta `sideMargin`, akurat neviem ako to vyzera na X

marekdovjak replied:
done

jcaliksygic commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -56,25 +98,74 @@ extension SYMKBrowseMapViewController: SYMapViewDelegate {
+        var viewObj: SYViewObject?
+        
+        for obj in objects {
+            if let poi = obj as? SYPoiObject, poi.type == .poi, mapSelectionMode == .all || mapSelectionMode == .all {
```
```suggestion
            if let poi = obj as? SYPoiObject, poi.type == .poi, mapSelectionMode == .all {
```

Kraci commented on SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift:
```diff
@@ -17,15 +40,19 @@ public class SYMKBrowseMapViewController: UIViewController {
             view.compass.viewModel = SYUICompassViewModel(course: 0, autoHide: true)
             view.compass.isHidden = !useCompass
+            view.recenter.setup(with: ActionButtonViewModel(title: "", icon: SygicIcon.positionLockIos, style: .secondary))
+            view.recenter.isHidden = !useRecenterButton
```
Tu ma teraz napadlo to napisat trosku krajsie ako som to pripravil..
Najprv si vytvorit `let browseView = SYMKBrowserMapView()`
nastavit mu postupne tieto properties a potom uz len zavolat `view = browserView`

jcaliksygic committed:
Update SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift

Co-Authored-By: marekdovjak <mdovjak@sygic.com>

jcaliksygic committed:
Update SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift

Co-Authored-By: marekdovjak <mdovjak@sygic.com>

jcaliksygic committed:
Update SygicMapsKit/Classes/BrowseMap/SYMKBrowseMapViewController.swift

Co-Authored-By: marekdovjak <mdovjak@sygic.com>

Commit:
PR fixes

Kraci committed:
Default view change moved to viewLoad()

- SYMKBrowserMapView renamed on SYMKBrowseMapView
- default view change in SYMKBrowseMapViewController moved to viewLoad() method
